src/rust/engine/src/externs/interface.rs

```diff
@@ -1626,19 +1626,27 @@ fn write_digest(
   py_session: &PySession,
   digest: &PyAny,
   path_prefix: String,
+  clear_destination: Option<bool>,
 ) -> PyO3Result<()> {
   let core = &py_scheduler.0.core;
   core.executor.enter(|| {
     // TODO: A parent_id should be an explicit argument.
     py_session.0.workunit_store().init_thread_state(None);
 
     let lifted_digest = nodes::lift_directory_digest(digest).map_err(PyValueError::new_err)?;
+    let clear_destination = clear_destination.unwrap_or(false);
```

Comment:
Since you've declared the function signature as requiring this argument, you don't need to make it optional.
```suggestion
  clear_destination: bool,
) -> PyO3Result<()> {
  let core = &py_scheduler.0.core;
  core.executor.enter(|| {
    // TODO: A parent_id should be an explicit argument.
    py_session.0.workunit_store().init_thread_state(None);

    let lifted_digest = nodes::lift_directory_digest(digest).map_err(PyValueError::new_err)?;
```